src/attr.rs

```diff
@@ -217,9 +217,10 @@ fn rewrite_initial_doc_comments(
     context: &RewriteContext<'_>,
     attrs: &[ast::Attribute],
     shape: Shape,
-) -> Option<(usize, Option<String>)> {
+) -> Result<(usize, Option<String>), RewriteError> {
+    // TODO Q. should we modify Option<String>
```

Comment:
I decided not to return replace Option<String> with RewriteResult, since the caller immediately panics on None. Though, I might have to revisit where rustfmt handles error with `panic` and `catch_unwind`.
```rust
// in attr.rs
let (doc_comment_len, doc_comment_str) = rewrite_initial_doc_comments(context, attrs, shape)?;
  if doc_comment_len > 0 {
    let doc_comment_str = doc_comment_str.expect("doc comments, but no result");
```